x-pack/plugin/eql/qa/common/src/main/resources/test_sample.toml

```diff
@@ -250,3 +250,61 @@ expected_event_ids  = [18,11]
 join_keys = ["doom","win10"]
 
 
+[[queries]]
+name = "size0"
+size = 0
```

Comment:
Other eql queries use `size` as well, but none of them require the toml tests to have `size` in them. I'm wondering if there isn't any other way to test `size` given that so far there was no need for `size` in IT toml tests.

Reply:
> For example, the Mapper is the one that generates physical plans based on the logical ones and, according to what's in there already a LimitWithOffset should map to a LimitWithOffsetExec. In your case, SampleExec physical plan includes a Limit, but it should in fact belong to a physical plan having as parent a LimitWithOffsetExec.

I'm not sure I got this comment: `LimitWithOffset+Sample` logical plan is transformed to `LimitWithOffsetExec+SampleExec`, then `QueryFolder.FoldLimit` rule applies the limit (from LimitWithOffsetExec) to SampleExec, that is then used to build SampleIterator.
This is the exact same thing that happens for Sequence execution planning.

About "size" in the tests, we have unit tests (mostly for the execution planning, plus some specific tests for the algorithms) in Java, but they do not fully execute any query (mocking the full execution is extremely convoluted due to the usage of internal - Server - components in the execution, that cannot be instantiated in our tests). 
The only real execution tests we have are TOML tests, this is why I added a size attribute there. It allows us to test the size on a full execution path.
In addition, I think also Sequence tests can take advantage of it.